src/app/system/manage-surveys/create-survey/create-survey.component.html

```diff
@@ -0,0 +1,161 @@
+
+            <div fxFlex="98%" fxLayout="row wrap" fxLayoutGap="2%" fxLayoutAlign="space-between center">
+              <h2 class="mat-h2">Question {{ questionIndex + 1 }}</h2>
+              <button mat-raised-button color="warn" (click)="removeQuestion(questionIndex)" [disabled]="questionDatas.controls.length === 1">
+                <fa-icon icon="trash"></fa-icon>&nbsp;&nbsp;
+                Delete Question
+              </button>
+            </div>
+
+            <mat-form-field fxFlex="48%">
+              <mat-label>Key</mat-label>
+              <input matInput maxlength="32" required formControlName="key">
+              <mat-error *ngIf="question.controls.key.hasError('required')">
+                Key is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+            
+            <mat-form-field fxFlex="48">
+              <mat-label>Text</mat-label>
+              <input matInput maxlength="255" required formControlName="text">
+              <mat-error *ngIf="question.controls.text.hasError('required')">
+                Text is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field fxFlex="98%">
+              <mat-label>Description</mat-label>
+              <textarea matInput formControlName="description"></textarea>
+            </mat-form-field>
+
+            <mat-divider id="margined-divider" [inset]="true"></mat-divider>
+
+            <div fxFlex="98%" fxLayout="row" fxLayoutAlign="space-between center">
+              <h4 class="mat-h4">Options</h4>
+              <button type="button" mat-raised-button color="primary" (click)="addResponse(questionIndex)">
+                <fa-icon icon="plus"></fa-icon>&nbsp;&nbsp;
+                Add Option
+              </button>
+            </div>
+
+            <div fxFlexFill formArrayName="responseDatas" *ngFor="let response of getResponseDatas(questionIndex).controls; let responseIndex = index;">
+              
+              <div fxFlexFill fxLayout="row wrap" fxLayoutGap="2%" [formGroupName]="responseIndex">
+
+                <mat-form-field fxFlex="43%">
+                  <mat-label>Text</mat-label>
+                  <input matInput required formControlName="text">
+                  <mat-error *ngIf="response.controls.text.hasError('required')">
+                    Text is <strong>required</strong>
+                  </mat-error>
+                </mat-form-field>
+
+                <mat-form-field fxFlex="43%">
+                  <mat-label>Value</mat-label>
+                  <input matInput required formControlName="value">
+                  <mat-error *ngIf="response.controls.value.hasError('required')">
+                    Value is <strong>required</strong>
+                  </mat-error>
+                  <mat-error *ngIf="response.controls.value.hasError('pattern')">
```

Comment:
I am aware that in other components, this would be getResponseDatas(questionIndex).at(responseIndex)..., but I do not see the point, as this is much more readable, and would not require the function to keep running over and over again.